### PR TITLE
Adding A log presence check before trying to delete

### PR DESCRIPTION
**Pull request by Anipik**
Related to https://github.com/dotnet/runtime/issues/36135

I run the tests in a loop, i am not able to repro this on the local machine

The error is we are trying to delete the non-existent log while clean up. So just adding a log check

### CODE REVIEW

**Review by ViktorHofer — COMMENTED** [review]
What about the other tests which call `EventLog.Delete` that you aren't guarding?

File: src/libraries/System.Diagnostics.EventLog/tests/EventLogTests/EventLogSourceCreationTests.cs [hunk 1]
```diff
@@ -23,13 +21,13 @@ public void CheckSourceExistenceAndDeletion()
             finally
             {
                 EventLog.DeleteEventSource(source);
-                Helpers.Retry(() => EventLog.Delete(log));  // unlike other tests, throw if delete fails
+                if(EventLog.Exists(log))
```
Comment (krwq) [inline src/libraries/System.Diagnostics.EventLog/tests/EventLogTests/EventLogSourceCreationTests.cs:24]:
```suggestion
                if (EventLog.Exists(log))
```

File: src/libraries/System.Diagnostics.EventLog/tests/EventLogTests/EventLogSourceCreationTests.cs [hunk 2]
```diff
@@ -23,13 +21,13 @@ public void CheckSourceExistenceAndDeletion()
...
+                    Helpers.Retry(() => EventLog.Delete(log));
```
Comment (krwq) [inline src/libraries/System.Diagnostics.EventLog/tests/EventLogTests/EventLogSourceCreationTests.cs:25]: I know this was here before but just curious why is this being retried?

**Comment by Anipik** [comment]
> What about the other tests which call EventLog.Delete that you aren't guarding?

We haven`t seen these failures in any other tests. We would need a repro before we can pin down "why only these tests."

File: src/libraries/System.Diagnostics.EventLog/tests/EventLogTests/EventLogSourceCreationTests.cs (hunk 2 above)
Comment (Anipik) [reply src/libraries/System.Diagnostics.EventLog/tests/EventLogTests/EventLogSourceCreationTests.cs:25]: Eventlog delete operation changes the registry and we require locks for that.  As in our tests, multiple threads are trying to achieve the same, sometimes it fails while waiting.

**Comment by Anipik** [comment]
@ViktorHofer @ericstj @safern can you take a look at this ?

**Review by safern — APPROVED** [review]
Looks good. We will have to be at the lookout to see if these fail again

**Pull request merged**